src/main.rs

```diff
@@ -773,17 +794,14 @@ fn get_setup(args: &[String]) -> Setup {
     let mixer_type: Option<String> = None;
 
     let mixer = mixer::find(mixer_type.as_deref()).unwrap_or_else(|| {
-        error!(
-            "Invalid `--{}` / `-{}`: {}",
+        invalid_error_msg(
             MIXER_TYPE,
             MIXER_TYPE_SHORT,
-            opt_str(MIXER_TYPE).unwrap_or_default()
-        );
-        println!(
-            "Valid `--{}` / `-{}` values: alsa, softvol",
-            MIXER_TYPE, MIXER_TYPE_SHORT
+            &opt_str(MIXER_TYPE).unwrap_or_default(),
+            "alsa, softvol",
```

Comment:
This depends on whether Alsa is compiled in or not.

Reply:
If alsa is not compiled in a user will never see that message because above that:

```rust
    #[cfg(feature = "alsa-backend")]
    let mixer_type = opt_str(MIXER_TYPE);
    #[cfg(not(feature = "alsa-backend"))]
    let mixer_type: Option<String> = None;    
```

Reply:
Above that:

```rust
    #[cfg(not(feature = "alsa-backend"))]
    for a in &[
        MIXER_TYPE,
        ALSA_MIXER_DEVICE,
        ALSA_MIXER_INDEX,
        ALSA_MIXER_CONTROL,
    ] {
        if opt_present(a) {
            warn!("Alsa specific options have no effect if the alsa backend is not enabled at build time.");
            break;
        }
    }
```

Reply:
If alsa is not compiled in that alsa specific options are short circuited and not evaluated.

Reply:
> If alsa is not compiled in a user will never see that message because above that:
> 
> ```rust
>     #[cfg(feature = "alsa-backend")]
>     let mixer_type = opt_str(MIXER_TYPE);
>     #[cfg(not(feature = "alsa-backend"))]
>     let mixer_type: Option<String> = None;    
> ```

OK, I missed that.

Reply:
It's Ok it's like a 1500 line function. There's a lot going on,lol!!!